packages/collector/test/tracing/cloud/azure/blob/test.js

```diff
@@ -0,0 +1,477 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+const { v4: uuid } = require('uuid');
+const expect = require('chai').expect;
+const semver = require('semver');
+const constants = require('@instana/core').tracing.constants;
+const supportedVersion = require('@instana/core').tracing.supportedVersion;
+const config = require('../../../../../../core/test/config');
+const { verifyHttpRootEntry, expectExactlyNMatching } = require('../../../../../../core/test/test_util');
+const ProcessControls = require('../../../../test_util/ProcessControls');
+const globalAgent = require('../../../../globalAgent');
+const { fail } = expect;
+const testUtils = require('../../../../../../core/test/test_util');
+const { createContainer, deleteContainer, minimumNodeJsVer } = require('./util');
+const { BlobServiceClient } = require('@azure/storage-blob');
+const containerName = `nodejs-team-${uuid()}`;
+const storageAccount = process.env.AZ_STORAGE_ACCOUNT_NAME;
+const accountKey = process.env.AZ_STORAGE_ACCOUNT_KEY;
+
+/**
+ * endPoint is assigned with the default constant string for appending EndpointSuffix
+ * to connection string of an azure account
+ */
+const endPoint = 'EndpointSuffix=core.windows.net';
+const connStr = `DefaultEndpointsProtocol=https;AccountName=${storageAccount};AccountKey=${accountKey};${endPoint}`;
+const blobServiceClient = BlobServiceClient.fromConnectionString(connStr);
+const containerClient = blobServiceClient.getContainerClient(containerName);
+
+let mochaSuiteFn = supportedVersion(process.versions.node) ? describe : describe.skip;
+mochaSuiteFn = semver.lt(process.versions.node, minimumNodeJsVer) ? describe.skip : mochaSuiteFn;
+
+/**
+ * This suite is skipped if no storageAccount or accountKey has been provided via AZ_STORAGE_ACCOUNT_NAME
+ * and AZ_STORAGE_ACCOUNT_KEY. For the Azure blob tests, the azure storage account used is teamnodejstracer
+ * which is the value for AZ_STORAGE_ACCOUNT_NAME. From the azure portal, navigate to this storage account
+ * and under the Access keys, Key can be found for AZ_STORAGE_ACCOUNT_KEY.
+ */
+if (!storageAccount || !accountKey) {
+  describe('tracing/cloud/azure/blob', function () {
+    it('The configuration for Azure is missing', () => {
+      fail('Please set process.env.AZ_STORAGE_ACCOUNT_KEY and process.env.AZ_STORAGE_ACCOUNT_NAME before tests.');
+    });
+  });
+} else {
+  mochaSuiteFn('tracing/cloud/azure/blob', function () {
+    globalAgent.setUpCleanUpHooks();
+    const agentControls = globalAgent.instance;
+
+    const controls = new ProcessControls({
+      dirname: __dirname,
+      useGlobalAgent: true,
+      env: {
+        AZURE_CONTAINER_NAME: containerName,
+        AZURE_CONNECTION_STRING: connStr,
+        AZURE_STORAGE_ACCOUNT: storageAccount,
+        AZURE_ACCOUNT_KEY: accountKey
+      }
+    });
+    ProcessControls.setUpHooks(controls);
+    this.timeout(config.getTestTimeout());
```

Comment:
The configuration of a custom timeout always needs to be on the first line.
The `ProcessControls` are registering before and after hooks as well. If we do not increase the timeout in the first line, they can easily run into the default timeout (which is two seconds).
This timeout can happen when running code coverage tools, which wrap our tests.